lib/ui.coffee

```diff
@@ -15,14 +17,15 @@ module.exports =
     @subs.add @client.onDetached =>
       @ink?.Result.invalidateAll()
 
-    @client.handle progress: (p) =>
-      @progress?.progress = p
+    @client.handle 'progress!': (t, p) => @progress[t] p
```

Comment:
The convention for these handlers is to use js-style naming since it works well in both languages